components/service/glean/src/main/java/mozilla/components/service/glean/private/EventMetricType.kt

```diff
@@ -62,14 +63,10 @@ data class EventMetricType<ExtraKeysEnum : Enum<ExtraKeysEnum>>(
             // There are two extra "keys" in play here:
             //   1. The Kotlin enumeration names, in CamelCase
             //   2. The keys sent in the ping, in snake_case
-            // Here we need to get (2) to send in the ping.  We need to use reflection to get
-            // around limitations in Java generics (since the actual enums for the event are in
-            // a template parameter).
+            // Here we need to get (2) to send in the ping.
             if (it.size > 0) {
-                val cls = it.keys.first().javaClass
-                val method = cls.getDeclaredMethod("getValue")
                 it.mapKeys { entry ->
-                    method.invoke(entry.key).toString()
+                    allowedExtraKeys[entry.key.ordinal]
```

Comment:
This is highly unlikely (it would mean a bug on our end), but what if `allowedExtraKeys` is empty? Would this be a NPE? Should we deal with that?

Reply:
That's a good point.  At a minimum we should avoid the NPE, but I don't know if we have to do anything much more sophisticated.

Reply:
Yes, nothing fancy. I think using [`elemetnAtOrElse`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/element-at-or-else.html) and printing an error should be enough.